packages/http/src/mocker/negotiator/NegotiatorHelpers.ts

```diff
@@ -206,16 +206,16 @@ const helpers = {
   negotiateOptionsBySpecificCode(
     httpOperation: IHttpOperation,
     desiredOptions: NegotiationOptions,
-    code: string
+    code: number
   ): RE.ReaderEither<Logger, Error, IHttpNegotiationResult> {
     // find response by provided status code
     return pipe(
       withLogger(logger =>
         pipe(
-          findResponseByStatusCode(httpOperation.responses, code),
+          findResponseByStatusCode(httpOperation.responses, code.toString()),
```

Comment:
I think these functions should still take a number; the conversion should be done internally.

Reply:
@XVincentX These functions took a string before, so this would be changing them to take a number, and do the conversion internally. But that can be done.

If I do that I have to change [here](https://github.com/stoplightio/prism/blob/master/packages/http/src/mocker/negotiator/NegotiatorHelpers.ts#L272), [here](https://github.com/stoplightio/prism/blob/master/packages/http/src/mocker/negotiator/NegotiatorHelpers.ts#L277), and [here](https://github.com/stoplightio/prism/blob/master/packages/http/src/mocker/negotiator/NegotiatorHelpers.ts#L269) to convert the `first` and `current` variables to a number. Or trace all the way back to where those are passed in and change those to numbers along the way. Worth it?

Reply:
Generally speaking, I want Prism to have a consistent interface, no matter the cost.

I'm obviously willing to sacrifice it in case it'd be such a pain, but if so it's usually because of an underlying design issue that should be fixed somehow.

Let me take a look into the code and I'll get back to you.

Reply:
ok cool. I'll look into all the things we'd need to change too.

Reply:
[this](https://github.com/stoplightio/prism/blob/master/packages/http/src/mocker/negotiator/InternalHelpers.ts#L62) is gonna keep us from changing the code param to a number for the `findResponseByStatusCode`.

Reply:
I figured, and that function looks wrong. Check https://github.com/stoplightio/prism/pull/1544 out.

Reply:
done. ea2ae13